source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs

```diff
@@ -102,12 +102,19 @@ public void Update(ChargePeriod newChargePeriod)
                 throw new InvalidOperationException("Charge update must not have bound end date.");
             }
 
+            var stopPeriod = _periods.FirstOrDefault(p => p.StartDateTime == p.EndDateTime
```

Comment:
There should always be 0 or 1 stop period, so I suggest:
```suggestion
            var stopPeriod = _periods.SingleOrDefault(p => p.StartDateTime == p.EndDateTime
```